my-code/src/app.js

```diff
@@ -0,0 +1,170 @@
+import Home from './views/home'
+import Movie from './views/movie'
+import './app.scss'
+
+class App extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      showcase: [],
+      searchEmpty: '',
+      loaderShowcase: 'hide',
+      movieEmpty: 'hide',
+      statusMovie: 'hide',
+      statusEmpty: 'hide',
+      statusWelcome: 'show',
+      movie: {},
+      favourites: [],
+      disableFavourites: false
+    }
+    this.handleSearch = (e) => {
+      e.preventDefault()
+      const value = e.target.search.value
+      this.setState({
+        loaderShowcase: 'show',
+        showcase: [],
+        movieEmpty: 'hide',
+        statusWelcome: 'hide'
+      })
+      if (value === '') {
+        this.setState({
+          loaderShowcase: 'hide',
+          searchEmpty: 'error',
+          movieEmpty: 'hide'
+        })
+        return
+      }
+      fetch(`http://www.omdbapi.com/?type=movie&s=${value}${process.env.API_IMDB}`)
+        .then(response => response.json())
+        .then((data) => {
+          const status = data.Response
+          const result = data.Search
+          if (status === 'False') {
+            this.setState({
+              searchEmpty: '',
+              loaderShowcase: 'hide',
+              movieEmpty: 'show'
+            })
+            return
+          }
+          this.setState({
+            searchEmpty: '',
+            movieEmpty: 'hide',
+            showcase: result,
+            loaderShowcase: 'hide'
+          })
+        })
+    }
+    this.handleGetMovie = (id) => () => {
+      const imdbID = id
+      let isFavourites = ''
+      const favourites = JSON.parse(localStorage.getItem('favourites'))
+      if (favourites !== null) {
+        isFavourites = favourites.filter((item) => item.imdb === imdbID)
+      }
+      fetch(`http://www.omdbapi.com/?i=${imdbID}${process.env.API_IMDB}`)
+        .then(response => response.json())
+        .then((data) => {
+          if (data.Response === 'False') {
+            this.setState({
+              statusEmpty: 'show'
+            })
+            return
+          }
+          if (isFavourites.length > 0) {
+            this.setState({
+              disableFavourites: true
+            })
+          }
+          this.setState({
+            movie: data,
+            statusMovie: 'show',
+            statusEmpty: 'hide'
+          })
+        })
+    }
+    this.handleBack = () => {
+      this.setState({
+        movie: {},
+        showcase: [],
+        statusMovie: 'hide',
+        statusEmpty: 'hide',
+        statusWelcome: 'show',
+        disableFavourites: false
+      })
+    }
+    this.handleFavourites = (id) => () => {
+      const newFavourite = this.state.favourites
+      newFavourite.push({'imdb': id})
+      this.setState({
+        disableFavourites: true,
+        favourites: newFavourite
+      })
+      localStorage.setItem('favourites', JSON.stringify(this.state.favourites))
```

Comment:
You could handle this in `componentDidUpdate`, so every new favorite update you can set it in `localStorate`